tests/fixtures_new.py

```diff
@@ -0,0 +1,240 @@
+import pytest
+from tests import helpers
+from ocs_ci.ocs import constants, ocp
+
+
+@pytest.fixture()
+def secret(request, interface_type):
+    """
+    Create a secret
+
+    Returns:
+        OCS: An instance of secret OCS
+
+    """
+    secrets = list()
+
+    def finalizer():
+        """
+        Delete the RBD secret
+        """
+        for secret in secrets:
+            secret.delete()
+            secret.ocp.wait_for_delete(secret.name)
+
+    request.addfinalizer(finalizer)
+    secrets.append(helpers.create_secret(interface_type=interface_type))
+
+    return secrets
```

Comment:
It should return single resource here I guess as have single name in the fixture, look at Filips pr how those fixture factories are defined if you would like to achieve something similar.